sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/aio/_operations/_patch.py

```diff
@@ -0,0 +1,229 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+
+import asyncio
+from typing import Any, Callable, IO, Coroutine, List, Optional, Union, cast
+
+from azure.core.polling import AsyncLROPoller, AsyncNoPolling, AsyncPollingMethod
+
+from azure.confidentialledger.aio._operations._operations import (
+    ConfidentialLedgerClientOperationsMixin as GeneratedOperationsMixin,
+)
+from azure.confidentialledger.aio._operations._operations import JSON
+
+__all__: List[str] = [
+    "ConfidentialLedgerClientOperationsMixin"
+]  # Add all objects you want publicly available to users at this package level
+
+
+def patch_sdk():
+    """Do not remove from this file.
+
+    `patch_sdk` is a last resort escape hatch that allows you to do customizations
+    you can't accomplish using the techniques described in
+    https://aka.ms/azsdk/python/dpcodegen/python/customize
+    """
+
+
+class StatePollingMethod(AsyncPollingMethod):
+    """Polling method for methods returning responses containing a 'state' field; the polling
+    completes when 'state' becomes a desired value.
+    """
+
+    def __init__(
+        self,
+        operation: Callable[[], Coroutine[Any, Any, JSON]],
+        desired_state: str,
+        polling_interval_s: float,
+    ):
+        self._operation = operation
+        self._desired_state = desired_state
+        self._polling_interval_s = polling_interval_s
+
+    def initialize(self, client, initial_response, deserialization_callback):
+        self._evaluate_response(initial_response)
+        self._deserialization_callback = deserialization_callback
+
+    def _evaluate_response(self, response: JSON) -> None:
+        self._status = "finished" if response["state"] == self._desired_state else "polling"
+        self._latest_response = response
+
+    async def run(self) -> None:
+        try:
+            while not self.finished():
+                response = await self._operation()
+                self._evaluate_response(response)
+
+                if not self.finished():
+                    await asyncio.sleep(self._polling_interval_s)
+        except Exception:
+            self._status = "failed"
+            raise
+
+    def status(self) -> str:
+        return self._status
+
+    def finished(self) -> bool:
+        return self.status() in {"finished", "failed"}
+
+    def resource(self):
```

Comment:
implementation suggestion: if you want, you can also make base classes in the sync patch.py that share the sansIO code, then you can have a sync and async class inherit from that class so you don't have to repeat code. This applies to clients,  operation groups etc

Reply:
Did this, it was gnarly, please take a look!